File: src/types/flow.ts
export type FlowCategory =
  | "work"
  | "family"
  | "home"
  | "wellness"
  | "play"
  | "growth";

export type FlowTaskType = "priority" | "chore" | "flex";

export type FlowTaskStatus =
  | "pending"
  | "in_progress"
  | "done"
  | "skipped"
  | "failed";

export type FlowTask = {
  id: string;
  title: string;
  type: FlowTaskType;
  category: FlowCategory;
  estimateMinutes: number;
  sequence: number;
  locked?: boolean;
  templateId?: string | null;
  scheduledStart?: string | null;
  scheduledEnd?: string | null;
  status: FlowTaskStatus;
  notes?: string | null;
  actualStart?: string | null;
  actualEnd?: string | null;
  createdAt: string;
  updatedAt: string;
};

export type FlowReflectionSentiment = "positive" | "neutral" | "challenging";

export type FlowReflection = {
  id: string;
  taskId: string | null;
  note: string;
  sentiment: FlowReflectionSentiment;
  mood?: string | null;
  moodLabel?: string | null;
  photoUrl?: string | null;
  createdAt: string;
};

export type FlowPlan = {
  id: string;
  date: string;
  timezone: string;
  startTime: string;
  autoScheduleEnabled?: boolean;
  tasks: FlowTask[];
  reflections: FlowReflection[];
  createdAt: string;
  updatedAt: string;
};

export type FlowMealPreference = {
  id: "lunch" | "dinner" | string;
  label: string;
  time: string;
  durationMinutes: number;
};

export type FlowFixedEventPreference = {
  id: string;
  label: string;
  category: FlowCategory;
  startTime: string;
  durationMinutes: number;
  days?: FlowDayOfWeek[];
  tags?: string[];
};

export type FlowDayOfWeek =
  | "monday"
  | "tuesday"
  | "wednesday"
  | "thursday"
  | "friday"
  | "saturday"
  | "sunday";

export const FLOW_DAY_ORDER: FlowDayOfWeek[] = [
  "monday",
  "tuesday",
  "wednesday",
  "thursday",
  "friday",
  "saturday",
  "sunday",
];

export type FlowSleepOverrides = Partial<
  Record<
    FlowDayOfWeek,
    {
      sleepStart: string;
      sleepEnd: string;
    }
  >
>;

export type FlowSettings = {
  workStart: string;
  workEnd: string;
  sleepStart: string;
  sleepEnd: string;
  meals: FlowMealPreference[];
  fixedEvents: FlowFixedEventPreference[];
  sleepOverrides?: FlowSleepOverrides;
  updatedAt: string;
  timezone: string;
};

File: android/toodl-share/app/src/main/java/com/toodl/share/data/model/Flow.kt
package com.toodl.share.data.model

data class FlowPlan(
    val id: String = "",
    val date: String = "",
    val timezone: String = "",
    val startTime: String = "",
    val autoScheduleEnabled: Boolean = false,
    val tasks: List<FlowTask> = emptyList(),
    val reflections: List<FlowReflection> = emptyList(),
    val createdAt: String = "",
    val updatedAt: String = ""
)

data class FlowTask(
    val id: String = "",
    val title: String = "",
    val type: String = "flex", // priority, chore, flex
    val category: String = "work", // work, family, home, wellness, play, growth
    val estimateMinutes: Int = 0,
    val sequence: Int = 0,
    val locked: Boolean = false,
    val templateId: String? = null,
    val scheduledStart: String? = null,
    val scheduledEnd: String? = null,
    val status: String = "pending", // pending, in_progress, done, skipped, failed
    val notes: String? = null,
    val actualStart: String? = null,
    val actualEnd: String? = null,
    val createdAt: String = "",
    val updatedAt: String = ""
)

data class FlowReflection(
    val id: String = "",
    val taskId: String? = null,
    val note: String = "",
    val sentiment: String = "neutral", // positive, neutral, challenging
    val mood: String? = null,
    val moodLabel: String? = null,
    val photoUrl: String? = null,
    val createdAt: String = ""
)
